cobalt/browser/migrate_storage_record/migration_manager.h

```diff
@@ -15,12 +15,41 @@
 #ifndef COBALT_COBALT_MIGRATE_STORAGE_RECORD_MIGRATION_MANAGER_H_
 #define COBALT_COBALT_MIGRATE_STORAGE_RECORD_MIGRATION_MANAGER_H_
 
+#include <atomic>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "base/functional/bind.h"
+#include "content/public/browser/weak_document_ptr.h"
 #include "content/public/browser/web_contents.h"
+#include "net/cookies/canonical_cookie.h"
 
 namespace cobalt {
 namespace migrate_storage_record {
 
-void DoMigrationTasksOnce(content::WebContents* web_contents);
+// Used to sequence tasks.
+using Task = base::OnceCallback<void(base::OnceClosure)>;
+
+class MigrationManager {
+ public:
+  static void DoMigrationTasksOnce(content::WebContents* web_contents);
+
+ private:
+  friend class MigrationManagerTest;
+
+  // Returns a task. When invoked, the grouped tasks are run sequentially.
+  static Task GroupTasks(std::vector<Task> tasks);
```

Comment:
I wonder if the code in MigrationManager and its test would be simplified by using a OnceCallbackList https://source.corp.google.com/h/lbshell-internal/cobalt_src/+/main:base/callback_list.h;l=282.

Also please remember to update the PR description: are the changes in migration_manager.h/cc only moving methods/class methods to the .h to be used by testing, or are there changes?